dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -382,6 +382,58 @@ public ISKFunction SetAIConfiguration(CompleteRequestSettings settings)
             : this.Function.SetAIConfiguration(settings);
     }
 
+    public ISKFunction SetPreExecutionHook(Func<SKContext, string, SKContext>? preHook)
```

Comment:
Currently considering bringing those as extension methods instead. Thoughts?

Reply:
Since hooks are optional parameters in `SKFunction` and `Plan`, I think the methods to set them should be optional as well, when you implement `ISKFunction` interface. Extension methods should help to achieve that.

Reply:
Those can't be extension methods because both are setting internal fields in the SKFunction implementation, any thoughts?